spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -189,16 +192,30 @@ public StreamingWrite toStreaming() {
 
   private Distribution buildRequiredDistribution() {
     if (overwriteFiles) {
-      throw new UnsupportedOperationException("Copy-on-write operations are temporarily not supported");
+      DistributionMode distributionMode = rowLevelDistributionMode();
+      return SparkDistributionAndOrderingUtil.buildCopyOnWriteDistribution(table, rowLevelCommand, distributionMode);
     } else {
       DistributionMode distributionMode = writeConf.distributionMode();
       return SparkDistributionAndOrderingUtil.buildRequiredDistribution(table, distributionMode);
     }
   }
 
+  private DistributionMode rowLevelDistributionMode() {
+    switch (rowLevelCommand) {
+      case DELETE:
+        return writeConf.copyOnWriteDeleteDistributionMode();
+      case UPDATE:
+        return writeConf.updateDistributionMode();
+      case MERGE:
```

Comment:
Can we remove UPDATE and MERGE here?

Reply:
Will do a pass through the PR to remove such places.

Reply:
Removed whatever found.